conans/test/functional/toolchains/env/test_virtualenv_powershell.py

```diff
@@ -114,3 +114,53 @@ def test(self):
     assert "MYPS1!!!!" in client.out
     assert "MYVC_CUSTOMVAR1=PATATA1" in client.out
     assert "MYVC_CUSTOMVAR2=PATATA2" in client.out
+
+@pytest.mark.skipif(platform.system() != "Windows", reason="Requires Windows powershell")
+def test_vcvars():
+    client = TestClient()
+    conanfile = textwrap.dedent(r"""
+        from conan import ConanFile
+        from conan.tools.cmake import CMake, cmake_layout, CMakeToolchain, CMakeDeps
+        from conan.tools.env import VirtualBuildEnv
+
+        class Conan(ConanFile):
+           settings = "os", "compiler", "build_type", "arch"
+
+           generators = 'CMakeDeps', 'CMakeToolchain'
+
+           def layout(self):
+              cmake_layout(self)
+
+           def build(self):
+              cmake = CMake(self)
+              cmake.configure()
+              cmake.build()
+    """)
+    client.save({"conanfile.py": conanfile})
+    hello_cpp = textwrap.dedent(r"""
+        int main(int argc, char **argv) {
+            return 0;
+        }
+    """)
+    client.save({"hello.cpp": hello_cpp})
+    cmakelists = textwrap.dedent(r"""
+        cmake_minimum_required(VERSION 3.15)
+
+        project(hello-world LANGUAGES CXX)
+        message("C++ compiler: ${CMAKE_CXX_COMPILER}")
+
+        add_executable(hello hello.cpp)
+    """)
```

Comment:
These files can most likely be generated in 2 lines with ``gen_function_cpp`` and ``gen_cmakelists`` helpers.